aea/aea_builder.py

```diff
@@ -910,49 +936,142 @@ def get_connection_configuration(connection_id):
             ]
 
         return [
-            _load_connection(address, get_connection_configuration(connection_id))
+            self._load_connection(address, get_connection_configuration(connection_id))
             for connection_id in connections_ids
         ]
 
     def _load_and_add_protocols(self) -> None:
         for configuration in self._package_dependency_manager.protocols.values():
             configuration = cast(ProtocolConfig, configuration)
-            try:
-                protocol = Protocol.from_config(configuration)
-            except ModuleNotFoundError as e:
-                _handle_error_while_loading_component_module_not_found(configuration, e)
-            except Exception as e:
-                _handle_error_while_loading_component_generic_error(configuration, e)
+            if (
+                configuration
+                in self._component_instances[ComponentType.PROTOCOL].keys()
+            ):
+                protocol = self._component_instances[ComponentType.PROTOCOL][
+                    configuration
+                ]
+            else:
```

Comment:
we now first check if we have the component instance already in the builder, otherwise proceed to load from dir.